### PR TITLE
Fix parsing pipelines what use a string as the skip key in a matrix adjustment

### PR DESCRIPTION
**Pull request by moskyb**
Also, demote SoftFail on matrix adjustments to being a RemainingField, as it's unused by the agent

### CODE REVIEW

File: internal/pipeline/step_command_matrix.go [hunk 1]
```diff
@@ -176,11 +176,23 @@ type MatrixAdjustments []*MatrixAdjustment
...
+	RemainingFields map[string]any `yaml:",inline"` // NB: soft_fail is in the remaining fields
+}
+
+func (ma *MatrixAdjustment) ShouldSkip() bool {
+	switch s := ma.Skip.(type) {
+	case bool:
+		return s
+
+	case string:
+		return s != ""
```
Comment (triarius) [inline internal/pipeline/step_command_matrix.go:191]: Pretty sure the empty string is not falsy in ruby. So if the backend sees
```yaml
skip: ""
```
it will adjust to skip.

Unless, that is, if the backend is using `.emtpy?` or the like. But I don't think it is:
```ruby
      jobs.reject! do |job|
        if job.matrix >= adjustment.with
          job.soft_fail = adjustment.soft_fail

          # Return the skip value so `reject!` will remove matching elements
          adjustment.skip
        end
      end
```

File: internal/pipeline/steps.go [hunk 2]
```diff
@@ -142,11 +142,6 @@ func (s Steps) sign(env map[string]string, key jwk.Key) error {
 	for _, step := range s {
 		switch step := step.(type) {
 		case *CommandStep:
-			if step.Matrix != nil {
-				// Don't sign matrix steps... yet
-				continue
-			}
-
```
Comment (triarius) [inline internal/pipeline/steps.go:149]: Maybe do this in another PR?

File: internal/pipeline/step_command_matrix.go [hunk 3]
```diff
@@ -176,11 +176,23 @@ type MatrixAdjustments []*MatrixAdjustment
...
+
+	default:
+		return false
```
Comment (triarius) [inline internal/pipeline/step_command_matrix.go:194]: Does this mean things like
```
skip: 1
```
are equivalent to
```
skip: false
```

Not sure if that's what we want

**Review by triarius — COMMENTED** [review]
I've assumed `ShouldSkip` should have the same logic as what removes adjustments from the jobs array. I don't think it currently does.

File: internal/pipeline/steps.go (hunk 2 above)
Comment (moskyb) [reply internal/pipeline/steps.go:149]: oh yes, definitely, that slipped in. will remove.

File: internal/pipeline/step_command_matrix.go (hunk 1 above)
Comment (moskyb) [reply internal/pipeline/step_command_matrix.go:191]: yeah, i think you're right. in ruby, only `nil` and `false` are truthy, so i guess we should just check for those

**Review by triarius — APPROVED** [review]
LGTM!
